include/onnxruntime/core/framework/ortdevice.h

```diff
@@ -28,31 +30,42 @@ struct OrtDevice {
     static const MemoryType QNN_HTP_SHARED = 4;
   };
 
-  constexpr OrtDevice(DeviceType device_type_, MemoryType memory_type_, DeviceId device_id_)
+  constexpr OrtDevice(DeviceType device_type_, MemoryType memory_type_, DeviceId device_id_, Alignment alignment) noexcept
       : device_type(device_type_),
         memory_type(memory_type_),
-        device_id(device_id_) {}
+        device_id(device_id_),
+        alignment(alignment) {}
 
-  constexpr OrtDevice() : OrtDevice(CPU, MemType::DEFAULT, 0) {}
+  // OrtDevice propagates many units and calling Mlas inline would have to change the linkage for many
+  // libs. However, we should save on calling MlasGetPreferredBufferAlignment() everytime we Alloc()
+  OrtDevice(DeviceType device_type_, MemoryType memory_type_, DeviceId device_id_) noexcept
+      : OrtDevice(device_type_, memory_type_, device_id_, ::MlasGetPreferredBufferAlignment()) {}
```

Comment:
the MLAS alignment is only relevant to CPU though, so using it as a default for all device types might not be desirable.

but isn't that alignment mainly relevant to the CPU EP, so we could call MlasGetPreferredBufferAlignment as pass it into OrtDevice when constructing an allocator in a CPU based EP? any CPU based EP could override the allocator for default CPU memory (e.g. xnnpack) so if we're going to fallback to CPU EP we still want the alignment to apply.